test/type/number_test.exs

```diff
@@ -0,0 +1,140 @@
+defmodule Litmus.Type.NumberTest do
+  use ExUnit.Case
+
+  alias Litmus.Type
+
+  describe "validate_field/3" do
+    test "validates property values of data based on their Number schema definition in Type.Number module" do
+      field = "id"
+      data = %{"id" => 1}
+
+      type = %Type.Number{
+        required: true
+      }
+
+      assert Type.Number.validate_field(type, field, data) == {:ok, data}
+    end
+  end
+
+  describe "convert string to number" do
+    test "returns :ok with modified value when convert is true" do
+      data = %{"id" => ".6"}
+      data_1 = %{"id" => "6"}
+      modified_data = %{"id" => 0.6}
+      modified_data_1 = %{"id" => 6}
+
+      schema = %{
+        "id" => %Litmus.Type.Number{},
+        "token" => %Litmus.Type.Number{}
+      }
+
+      assert Litmus.validate(data, schema) == {:ok, modified_data}
+      assert Litmus.validate(data_1, schema) == {:ok, modified_data_1}
+    end
+
+    test "errors when convert is true and field type is neither number or stringified number" do
+      field = "id"
+      data = %{"id" => "1.a"}
+      data_1 = %{"id" => true}
+
+      schema = %{
+        "id" => %Litmus.Type.Number{}
+      }
+
+      assert Litmus.validate(data, schema) == {:error, "#{field} must be a number"}
+      assert Litmus.validate(data_1, schema) == {:error, "#{field} must be a number"}
```

Comment:
I think we can probably just make this `id must be a number` and remove the `field` assignment above.